numpy/ma/core.py

```diff
@@ -3315,7 +3319,10 @@ def _scalar_heuristic(arr, elem):
                                 "heterogeneous fill_value and setting "
                                 f"all to {dout._fill_value[0]!s}.",
                                 stacklevel=2)
-                        dout._fill_value = dout._fill_value.flat[0]
+                        # need to use squeeze to ensure the result is an array 
+                        # not a scalar. If the _fill_value is like a str type,
+                        # then just make sure we turn the _fill_value to a 0d array
+                        dout._fill_value = dout._fill_value.flat[0:1].squeeze(axis=0)
```

Comment:
This doesn't quite capture the problem here. Perhaps
```suggestion
                        # Need to use `.flat[0:1].squeeze(...)` instead of just
                        # `.flat[0]` to ensure the result is a 0d array and not
                        # a scalar.
                        dout._fill_value = dout._fill_value.flat[0:1].squeeze(axis=0)
```